pkg/manager/volumes/delegation/azure/disk_modifier.go

```diff
@@ -0,0 +1,308 @@
+		IOPS:       disk.Properties.DiskIOPSReadWrite,
+		Throughput: disk.Properties.DiskMBpsReadWrite,
+		Type:       string(*disk.SKU.Name),
+	}, nil
+}
+
+func (m *AzureDiskModifier) getExpectedVolume(pvc *corev1.PersistentVolumeClaim, pv *corev1.PersistentVolume, sc *storagev1.StorageClass) (*Volume, error) {
+	v := Volume{}
+	if err := utilerrors.NewAggregate([]error{
+		m.setArgsFromPVC(&v, pvc),
+		m.setArgsFromPV(&v, pv),
+		m.setArgsFromStorageClass(&v, sc),
+	}); err != nil {
+		return nil, err
+	}
+
+	return &v, nil
+}
+
+func (m *AzureDiskModifier) MinWaitDuration() time.Duration {
+	return defaultWaitDuration
+}
+
+func (m *AzureDiskModifier) setArgsFromPVC(v *Volume, pvc *corev1.PersistentVolumeClaim) error {
+	size, err := getSizeFromPVC(pvc)
+	if err != nil {
+		return err
+	}
+	v.Size = ptr.To(int32(size))
+	return nil
+}
+
+func getSizeFromPVC(pvc *corev1.PersistentVolumeClaim) (int64, error) {
+	quantity := pvc.Spec.Resources.Requests[corev1.ResourceStorage]
+	sizeBytes := quantity.ScaledValue(0)
+	size := sizeBytes / 1024 / 1024 / 1024
+
+	if size < minSize || size > maxSize {
+		return 0, fmt.Errorf("invalid storage size: %v", quantity)
+	}
+	return size, nil
+}
+
+func (m *AzureDiskModifier) setArgsFromPV(v *Volume, pv *corev1.PersistentVolume) error {
+	v.VolumeId = pv.Spec.CSI.VolumeHandle
+	return nil
+}
+
+func (m *AzureDiskModifier) setArgsFromStorageClass(v *Volume, sc *storagev1.StorageClass) error {
+	if sc == nil {
+		return nil
+	}
+	throughput, err := getParamInt64(sc.Parameters, paramKeyThroughput)
+	if err != nil {
+		return err
+	}
+	v.Throughput = throughput
+
+	iops, err := getParamInt64(sc.Parameters, paramKeyIOPS)
+	if err != nil {
+		return err
+	}
+	v.IOPS = iops
+
+	v.Type = sc.Parameters[paramKeyType]
+	return nil
+}
+
+func getParamInt64(params map[string]string, key string) (*int64, error) {
+	str, ok := params[key]
+	if !ok || str == "" {
+		return nil, nil
+	}
+	param, err := strconv.ParseInt(str, 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("can't parse %v param in storage class: %v", key, err)
+	}
+
+	return ptr.To(param), nil
+}
+
+func getDiskInfoFromVolumeID(volumeID string) (diskName string, subscriptionID string, resourceGroupName string, err error) {
+	// get diskName, subscriptionID, resourceGroupName from volumeHandle
+	// example: /subscriptions/xxxx/resourceGroups/xxxx/providers/Microsoft.Compute/disks/xxxx
+	parts := strings.Split(volumeID, "/")
+	if len(parts) < 9 {
+		return "", "", "", fmt.Errorf("invalid volumeHandle format")
+	}
+	subscriptionID = parts[2]
+	resourceGroupName = parts[4]
+	diskName = parts[len(parts)-1]
+
+	return diskName, subscriptionID, resourceGroupName, nil
+}
+
+func (m *AzureDiskModifier) setDiskClientForPV(subscriptionID string) error {
+	if m.DiskClient != nil {
+		return nil
+	}
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return fmt.Errorf("failed to obtain a credential: %v", err)
+	}
```

Comment:
What's the behavior without an Azure config? can we do the same check as we did for AWS in `dependences.go`?

Reply:
1. If there is no Azure config, it should not be initialized here, and no error will be reported. It's safe for non-Azure k8s environment.
2. Because the subscription and resource group that needs to be operated cannot be obtained from places such as Node Role, it has to be obtained from the disk handler of PV the subscription and resource group. For the same AKS and a tidb-controller-manager process, this information should be the same after it is obtained once, and it will only be initialized once in the process.